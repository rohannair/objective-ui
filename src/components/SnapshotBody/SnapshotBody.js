import React, { PropTypes } from 'react'
import styled from 'styled-components'

const SnapshotBody = p => {
  const { img, body, className } = p

  return (
    <div className={className}>
      {
        img && (
          <div className="imgContainer">
            <a href={img} target="_blank" rel="noopener">
              <img src={img} />
            </a>
          </div>
        )
      }
      { body && <div className="body" dangerouslySetInnerHTML={{ __html: body} }/>}
    </div>
  )
}

export default styled(SnapshotBody)`
  padding: 20px 0;
  .imgContainer {
    object-fit: cover;
    margin-bottom: 10px;
    overflow: hidden;
  }

  img {
    height: 260px;
    width: 100%;
    object-fit: cover;
  }

  .body {
    padding: 0 20px;
  }
`
