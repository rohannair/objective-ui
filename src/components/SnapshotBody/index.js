import SnapshotBody from './SnapshotBody'
export default SnapshotBody
