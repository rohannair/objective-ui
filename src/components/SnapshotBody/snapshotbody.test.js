import React from 'react'

import SnapshotBody from './index.js'

describe('Test suite for SnapshotBody component', () => {
  it('SnapshotBody should exist', () => {
    let wrapper = shallow(<SnapshotBody/>)
    expect(wrapper).toMatchSnapshot()
  })
})

