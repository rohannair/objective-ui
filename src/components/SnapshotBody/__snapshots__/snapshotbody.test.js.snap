exports[`Test suite for SnapshotBody component SnapshotBody should exist 1`] = `
<SnapshotBody
  className="cPnWJo" />
`;
